test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +91,11 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+
+        Command command = new Parser().parseCommand(inputCommand);
+        if(command.isMutating()) {
```

Comment:
Good effort! think about the negative test cases too. If isMutating has not been implemented this would pass too so think of other conditions you can capture to display the effects of having isMutating